libcni/api.go

```diff
@@ -72,8 +75,8 @@ type CNI interface {
 	GetNetwork(net *NetworkConfig, rt *RuntimeConf) (types.Result, error)
 	DelNetwork(net *NetworkConfig, rt *RuntimeConf) error
 
-	ValidateNetworkList(net *NetworkConfigList) ([]string, error)
-	ValidateNetwork(net *NetworkConfig) ([]string, error)
+	ValidateNetworkList(ctx context.Context, net *NetworkConfigList) ([]string, error)
```

Comment:
Per above, make AddNetwork/GetNetwork/DelNetowrk and their "List" functions do the same thing as ValidateNetwork[List]; just have callers pass a Context or nil.